administration/src/bp-modules/cards/AddCardForm.tsx

```diff
@@ -33,10 +33,17 @@ const ExtensionForm = ({ extension, onUpdate }: ExtensionFormProps) => {
 }
 
 const maxCardValidity = { years: 99 }
-const hasCardExpirationError = (expirationDate: PlainDate): boolean => {
+const hasCardExpirationError = (cardBlueprint: CardBlueprint): boolean => {
+  const expirationDate = cardBlueprint.expirationDate
+  if (!expirationDate) {
+    return false
+  }
   const today = PlainDate.fromLocalDate(new Date())
-
-  return expirationDate.isBefore(today) || expirationDate.isAfter(today.add(maxCardValidity))
+  return (
+    expirationDate.isBefore(today) ||
+    expirationDate.isAfter(today.add(maxCardValidity)) ||
+    !cardBlueprint.isExpirationDateValid()
+  )
```

Comment:
I think all of these checks can be moved into the CardBlueprint. Into the is ExpirationDateValid()

Reply:
moved please rereview :). @sarahsporck